src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1485,10 +1624,18 @@ private void ImportLoadString(int token)
 
         private void ImportInitObj(int token)
         {
+            TypeDesc type = ResolveTypeToken(token);
+            var valueEntry = _stack.Pop();
+            var llvmType = GetLLVMTypeForTypeDesc(type);
+            if (llvmType.TypeKind == LLVMTypeKind.LLVMArrayTypeKind)
+                LLVM.BuildStore(_builder, WebAssemblyObjectWriter.GetConstZeroArray(type.GetElementSize().AsInt), valueEntry.ValueAsType(LLVM.PointerType(llvmType, 0), this));
+            else
+                LLVM.BuildStore(_builder, BuildConstInt32(0), valueEntry.ValueAsType(LLVM.PointerType(LLVM.Int32Type(), 0), this));
```

Comment:
This could probably use some kind of assert since you seem to be assuming the only other possibility is that this is an int or pointer.